server/consumer.go

```diff
@@ -548,12 +551,7 @@ func (mset *Stream) AddConsumer(config *ConsumerConfig) (*Consumer, error) {
 // Do all advisory sends here.
 // Lock should be held on entry but will be released.
 func (o *Consumer) sendAdvisory(subj string, msg []byte) {
-	if o.mset != nil && o.mset.sendq != nil {
-		sendq := o.mset.sendq
-		o.mu.Unlock()
-		sendq <- &jsPubMsg{subj, subj, _EMPTY_, nil, msg, nil, 0}
-		o.mu.Lock()
-	}
+	o.sendq <- &jsPubMsg{subj, subj, _EMPTY_, nil, msg, nil, 0}
```

Comment:
As you know, I don't like releasing lock then reacquire in situations where the caller or function assumes that the state is not going to change. So happy to see above disappear, but at the same time very nervous about use of the go channel send under lock because the go routine reading may want to grab the lock at one point maybe? (can processInboundClientMsg() at one point end-up wanting to acquire the lock of this consumer?)

Reply:
Yes understand. In this case it can not since this is not a consumer delivered message. On consumer delivered messages because we process `did not deliver` status we can re-acquire the lock for the consumer in that case so deliverMsg will release the lock. 

For safe keeping can do so here as well even though not needed. Will make that change.